resources/lib/UnityLDAP.php

```diff
@@ -115,67 +115,58 @@ public function getDefUserShell()
   //
   // ID Number selection functions
   //
-    public function getNextUIDNumber()
+    public function getNextUIDNumber($UnitySQL)
     {
-        $users = $this->userOU->getChildrenArray(true);
+        $max_uid = $UnitySQL->getSiteVar('MAX_UID');
+        $new_uid = $max_uid + 1;
 
-      // This could become inefficient with more users
-        usort($users, function ($a, $b) {
-            return $a["uidnumber"] <=> $b["uidnumber"];
-        });
-
-        $id = self::ID_MAP[0];
-        foreach ($users as $acc) {
-            if ($id == $acc["uidnumber"][0]) {
-                $id++;
-            } else {
-                if (!$this->GIDNumInUse($id)) {
-                    break;
-                }
+        $is_already_in_use = $this->UIDNumInUse($new_uid);
+
+        if ($is_already_in_use) {
+            while ($this->UIDNumInUse($new_uid)) {
+                $new_uid++;
             }
         }
```

Comment:
```suggestion
       while ($this->UIDNumInUse($new_uid)) {
           $new_uid++;
       }
```

You only need the while loop here, if it's not in use it won't fire at all. This is the same for the other 2 as well.